enterprise/server/workflow/service/service.go

```diff
@@ -268,6 +274,142 @@ func (ws *workflowService) GetWorkflows(ctx context.Context, req *wfpb.GetWorkfl
 	return rsp, nil
 }
 
+func (ws *workflowService) ExecuteWorkflow(ctx context.Context, req *wfpb.ExecuteWorkflowRequest) (*wfpb.ExecuteWorkflowResponse, error) {
+	// Validate req
+	if req.GetWorkflowId() == "" {
+		return nil, status.InvalidArgumentError("Missing workflow_id")
+	}
+	if req.GetCommitSha() == "" {
+		return nil, status.InvalidArgumentError("Missing commit_sha")
+	}
+	if req.GetBranch() == "" {
+		return nil, status.InvalidArgumentError("Missing branch")
+	}
+	if req.GetActionName() == "" {
+		return nil, status.InvalidArgumentError("Missing action_name")
+	}
+
+	// Authenticate
+	user, err := perms.AuthenticatedUser(ctx, ws.env)
+	if err != nil {
+		return nil, err
+	}
+
+	// Lookup workflow
+	wf := &tables.Workflow{}
+	db := ws.env.GetDBHandle()
+	err = db.Raw(
+		`SELECT group_id, repo_url, access_token, perms FROM Workflows WHERE workflow_id = ?`,
+		req.GetWorkflowId(),
+	).Take(wf).Error
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
nit: please use IsRecordNotFound

(and remove any gorm imports)

Reply:
Done